src/utils.ts

```diff
@@ -1198,4 +1352,786 @@ export class RubberBand {
+    const path = new Path2D();
+    const endArrow = new newPoint2D(this.end.x + this.SIZE_END / 2 * notVert, this.end.y + this.SIZE_END / 2 * isVert, this.SIZE_END, 'triangle', ['right', 'up'][isVert]);
+    path.moveTo(this.origin.x - this.DRAW_START_OFFSET * notVert, this.origin.y - this.DRAW_START_OFFSET * isVert);
+    path.lineTo(this.end.x, this.end.y);
+    path.addPath(endArrow.path);
+    return path
+  }
+
+  private buildPath(): Path2D {
+    const path = new Path2D();
+    const offset = new Vertex(this.DRAW_START_OFFSET * Number(this.isVertical), this.DRAW_START_OFFSET * Number(!this.isVertical));
+    const origin = new Vertex(this.origin.x, this.origin.y).subtract(offset.multiply(2));
+    const size = this.end.subtract(origin).add(offset);
+    path.rect(origin.x, origin.y, size.x, size.y);
+    return path
+  }
+
+  public absoluteToRelative(value: number): number {
+    return this.isVertical ? (value - this.transformMatrix.f) / this.transformMatrix.d : (value - this.transformMatrix.e) / this.transformMatrix.a
+  }
+
+  public relativeToAbsolute(value: number): number {
+    return this.isVertical ? value * this.transformMatrix.d + this.transformMatrix.f : value * this.transformMatrix.a + this.transformMatrix.e
+  }
+
+  private computeMinMax(vector: any[]): number[] {
+    let newVector = vector;
+    if (this.isDiscrete) {return [0, this.labels.length]};
+    return [Math.min(...newVector), Math.max(...newVector)]
+  }
+
+  private computeTicks(): number[] {
+    const increment = newAxis.nearestFive((this.maxValue - this.minValue) / this.nTicks);
+    const remainder = this.minValue % increment;
+    let ticks = [this.minValue - remainder];
+    while (ticks.slice(-1)[0] <= this.maxValue) {ticks.push(ticks.slice(-1)[0] + increment)};
+    if (ticks.slice(0)[0] < this.minValue) {ticks.splice(0, 1)};
+    if (ticks.slice(-1)[0] > this.maxValue) {ticks.splice(-1, 1)};
+    return ticks
+  }
+
+  public draw(context: CanvasRenderingContext2D) {
+    context.lineWidth = this.lineWidth;
+    let color = this.strokeStyle;
+    if (this.mouseStyleON) {color = this.isHover ? this.hoverStyle : this.isClicked ? this.clickedStyle : this.strokeStyle};
+    context.strokeStyle = color;
+    context.fillStyle = color;
+    context.stroke(this.drawPath);
+    context.fill(this.drawPath);
+
+    const canvasHTMatrix = context.getTransform();
+    const pointHTMatrix = canvasHTMatrix.multiply(this.transformMatrix);
+    context.setTransform(pointHTMatrix);
+    this.ticksCoords = this.drawTicks(context, pointHTMatrix, color);
+    context.resetTransform();
+    this.drawName(context, canvasHTMatrix);
+    context.setTransform(canvasHTMatrix);
+    this.drawRubberBand(context);
+  }
+
+  private drawName(context: CanvasRenderingContext2D, canvasHTMatrix: DOMMatrix) {
+    let baseline = ['hanging', 'alphabetic'][Math.sign(1 - Math.sign(this.transformMatrix.f))]
```

Comment:
For increased readibility we should explicitly name what `Math.sign(1 - Math.sign(this.transformMatrix.f))` denotes

Moreover, it is defined twice, so I guess it could be interesting to set it in a variable

Reply:
It is defined only once and gives the index i need to use for choosing between hanging and alphabetic configuration.
Explicitly name it would add 2 code lines